### PR TITLE
Docs: improvements for "references - phpdoc - tags - property[<-read|-write>]"

### PR DESCRIPTION
**Pull request by jrfnl**
I've combined the contents of the `@property`, `@property-read` and `@property-write` pages into _one_ documentation page as it makes more sense to read about these tags together.
The added benefit is, that this removed some (near) duplication in the documentation.

Note:
* The pages will still show up individually in the table of contents, with the title for the `@property` documentation page being adjusted for the purposes of the TOC.
* Existing (external/internal) links to these pages will not be broken as the `@property-read` and `@property-write` pages still exist, they will now just show the same contents as the `@property` page.

Summary:
* Synced the text with the current text in PSR-19.
* Added a link to the PHP manual page covering class properties.

Syntax:
* Adjusted the specs to include `@property-read` and `@property-write`.

Description:
* Based the text on PSR-19, with a rewritten "example" from the original text.
* Added a table to make it more obvious when to use which tag.
* Added links to the PHP manual pages covering the relevant magic methods.
    As the link _text_ starts with an underscore, I wasn't sure having those links at the bottom of the page would work as there they would need to be prefixed with an underscore, so, in this case, for these specific links, I've left them inline.
* Rephrased the "MUST NOT" phrase which contained a double negative to use "can ONLY" for clarity.

Effects in phpDocumentor:
* Adjusted the text to include `@property-read` and `@property-write`.

Examples:
* Expanded the original example to cover all three tags.
* Added the example from PSR 19 as a secondary, more concrete example.

Other:
* Minor inline markup and grammar fixes.
* Made the syntax outline a code block.

Ref: https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc-tags.md#510-property

### CODE REVIEW

**Comment by mvriel** [comment]
@ashnazg would you mind reviewing this PR? I'm afraid my priorities are elsewhere at the moment but having this reviewed would be awesome :)

File: docs/references/phpdoc/tags/property.rst [hunk 1]
```diff
@@ -1,34 +1,49 @@
...
+============================ =====================
+``__get()`` and ``__set()``  ``@property``
+``__get()`` only             ``@property-read``
+``__set()`` only             ``@property-write``
+============================ =====================
 
-@property tags MUST NOT be used in a PHPDoc that is not associated with
-a *class* or *interface*.
+The ``@property``, ``@property-read`` and ``@property-write`` tags can ONLY be used
+in a PHPDoc that is associated with a *class* or *trait*.
```
Comment (ashnazg) [inline docs/references/phpdoc/tags/property.rst:38]: Add `interface` to this list?

Comment (jrfnl) [reply docs/references/phpdoc/tags/property.rst:38]: Eh... interfaces cannot declare properties, so these tags don't apply.

Comment (ashnazg) [reply docs/references/phpdoc/tags/property.rst:38]: Ah, it's _constants_ that I'm thinking of on interfaces.  Ok, then `*interface*` can be removed from line 43 below... or maybe _replaced_ by `*trait*`.

Comment (jrfnl) [reply docs/references/phpdoc/tags/property.rst:38]: Good catch, not sure how I missed that one. Fixed now.

**Review by ashnazg — APPROVED** [review]

**Pull request merged**